rules-engine/tests/rules_engine/test_examples.py

```diff
@@ -14,18 +18,97 @@
 INPUT_DATA = next(os.walk(ROOT_DIR))[1]
 
 
-def load_example(folder: str) -> dict[str, Any]:
+def validate_fuel_type(value):
+    try:
+        return engine.FuelType[value]
+    except KeyError as e:
+        raise ValueError(
+            f"Error validating fuel type {e}. Valid choices are: {[x.name for x in engine.FuelType]}"
+        )
+
+
+class Summary(BaseModel):
+    local_weather_station: str
+    design_temperature_override: Optional[float]
+    living_area: int
+    fuel_type: Annotated[engine.FuelType, BeforeValidator(validate_fuel_type)]
+    heating_system_efficiency: float
+    other_fuel_usage: Optional[float]
+    other_fuel_usage_override: Optional[float]
+    thermostat_set_point: float
+    setback_temp: float
+    setback_hours_per_day: float
+    estimated_balance_point: float
+    balance_point_sensitivity: float
+    average_indoor_temperature: float
+    difference_between_ti_and_tbp: float
+    design_temperature: float
+    whole_home_ua: int
+    standard_deviation_of_ua: float
+    avg_heat_load: int
+    max_heat_load: int
+
+
+def validate_datetime(value):
+    return datetime.strptime(value, "%m/%d/%Y").date()
+
+
+def validate_inclusion(value):
+    return int(value) if value else None
+
+
+class NaturalGasUsage(BaseModel):
+    start_date: Annotated[datetime, BeforeValidator(validate_datetime)]
+    end_date: Annotated[datetime, BeforeValidator(validate_datetime)]
+    days_in_bill: int
+    usage: int
+    inclusion_override: Annotated[
+        Optional[Literal[-1, 0, 1]], BeforeValidator(validate_inclusion)
+    ]
+    inclusion_code: Annotated[Literal[-1, 0, 1], BeforeValidator(validate_inclusion)]
+    avg_daily_usage: float
+    daily_htg_usage: float
+
+
+class Example(BaseModel):
+    summary: Summary
+    natural_gas_usage: Optional[List[NaturalGasUsage]]
```

Comment:
I wonder if this can be abstracted into a `fuel_usage` type for the other examples. Not necessary for this PR, but thoughts for later.

Reply:
I thought about this a bit too...

The format of data between natural gas and fuel oil is different, so there'd need to be something to account for that.  I could see this being either:

```py
class Example(BaseModel):
    summary: Summary
    natural_gas_usage: Optional[List[NaturalGasUsage]]
    fuel_oil_usage: Optional[List[FuelOilUsage]]
```

or alternatively:

```py
class Example(BaseModel):
    summary: Summary
    fuel_usage: Optional[List[NaturalGasUsage] | List[FuelOilUsage]]
```